### PR TITLE
Logs antagonist datum removal

### PR DESCRIPTION
**Pull request by Pepsilawn**
## About The Pull Request

This PR adds log_message logging for /datum/mind/proc/remove_antag_datum(), add_antag_datum was already properly logged but removal never was. This might end up resulting in some duplicate logging as a few antagonists have their own loggings for IC means of losing it, notably Revolutionary which broadcasts in attack.log mindshield deconversions with the deconverter's name, most don't though.

## Why It's Good For The Game

Antagonists that do not retain their status after being borged such as revolutionaries and cultists, obsessed and brainwashed loss through brain surgery, changelings absorbing another ling out of their antagonist status, cult soulshard being deconverted using a bible and more importantly cultist holy water deconversion are all logged now.

## Changelog

:cl:
admin: Added proper logging to antagonist datum removal
/:cl:

### CODE REVIEW

**Review by dragomagol — APPROVED** [review]

File: code/datums/mind/antag.dm [hunk 1]
```diff
@@ -33,13 +33,14 @@
 	var/datum/antagonist/A = has_antag_datum(datum_type)
 	if(A)
 		A.on_removal()
+		log_game("[key_name(src)] has lost antag datum [A.name]([A.type]).")
```
Comment (dragomagol) [inline code/datums/mind/antag.dm:36]:
```suggestion
		src.log_message("has lost antag datum [A.name]([A.type]).", LOG_GAME)
```

Comment (dragomagol) [reply code/datums/mind/antag.dm:36]: I believe that log_message is on mob, so you may have to do src.mob. Am mobileposting so I'm not 100%

Comment (Pepsilawn) [reply code/datums/mind/antag.dm]: Yeah it's an atom proc, passed it through current. instead and it seemed to work fine, cheers

File: code/datums/mind/antag.dm [hunk 2]
```diff
@@ -33,13 +33,14 @@
 	var/datum/antagonist/A = has_antag_datum(datum_type)
 	if(A)
 		A.on_removal()
+		current.log_message("[key_name(src)] has lost antag datum [A.name]([A.type]).", LOG_GAME)
```
Comment (dragomagol) [inline code/datums/mind/antag.dm:36]:
```suggestion
		current.log_message("has lost antag datum [A.name]([A.type]).", LOG_GAME)
```
log_message adds the name automatically when it logs to game.log

File: code/datums/mind/antag.dm [hunk 3]
```diff
@@ -33,13 +33,14 @@
...
 		A.on_removal()
+		current.log_message("has lost antag datum [A.name]([A.type]).", LOG_GAME)
 		return TRUE
 
-
 /datum/mind/proc/remove_all_antag_datums() //For the Lazy amongst us.
 	for(var/a in antag_datums)
 		var/datum/antagonist/A = a
 		A.on_removal()
+	current.log_message("[key_name(src)] has lost all antag datums.", LOG_GAME)
```
[the remark this answers is not in the document]
Comment (Pepsilawn) [reply code/datums/mind/antag.dm:43]: mb

**Pull request merged**